frontend/src/AudioAnnotator/Workbench.tsx

```diff
@@ -475,152 +479,161 @@ class Workbench extends Component<WorkbenchProps, WorkbenchState> {
 
   renderTimeAxis = () => {
     const timeAxis: HTMLCanvasElement = this.timeAxisRef.current;
-    const context: CanvasRenderingContext2D = timeAxis.getContext('2d');
-    context.clearRect(0, 0, timeAxis.width, timeAxis.height);
-
-    let step: number = 1; // step of scale (in seconds)
-    let bigStep: number = 5;
-
-    const durationOnScreen: number = this.state.wrapperWidth / this.state.timePxRatio;
-    if (durationOnScreen <= 60) {
-      step = 1;
-      bigStep = 5;
-    } else if (durationOnScreen > 60 && durationOnScreen <= 120) {
-      step = 2;
-      bigStep = 5;
-    } else if (durationOnScreen > 120 && durationOnScreen <= 500) {
-      step = 4;
-      bigStep = 5;
-    } else if (durationOnScreen > 500 && durationOnScreen <= 1000) {
-      step = 10;
-      bigStep = 60;
-    } else {
-      step = 30;
-      bigStep = 120;
-    }
-
-    const bounds: ClientRect = timeAxis.getBoundingClientRect();
-    const startTime: number = Math.ceil(this.getTimeFromClientX(bounds.left));
-    const endTime: number = Math.floor(this.getTimeFromClientX(bounds.right));
-
-    context.fillStyle = 'rgba(0, 0, 0)';
-    context.font = '10px Arial';
-
-    let i: number = 0;
-    for (i = startTime ; i <= endTime; i++) {
-      if (i % step === 0) {
-        const x: number = (i - startTime) * this.state.timePxRatio;
-
-        if (i % bigStep === 0) {
-          // Bar
-          context.fillRect(x, 0, 2, 15);
+    const context: CanvasRenderingContext2D | null = timeAxis.getContext('2d');
+
+    if (context) {
```

Comment:
to reduce the code complexity, use the early return pattern :
Instead of
```js
if (context) {
    ...
}
```
prefer
```js
if (!context) return;
...
```
(global comment, no need to change it now)

Reply:
Good remark, is is fixed in last commit.